go/tendermint/apps/staking/state/gas.go

```diff
@@ -0,0 +1,92 @@
+package state
+
+import (
+	"fmt"
+
+	"github.com/oasislabs/oasis-core/go/common/consensus/gas"
+	"github.com/oasislabs/oasis-core/go/common/crypto/signature"
+	"github.com/oasislabs/oasis-core/go/common/quantity"
+	staking "github.com/oasislabs/oasis-core/go/staking/api"
+	"github.com/oasislabs/oasis-core/go/tendermint/abci"
+)
+
+// feeAccumulatorKey is the block context key.
+type feeAccumulatorKey struct{}
+
+func (fak feeAccumulatorKey) NewDefault() interface{} {
+	return &feeAccumulator{}
+}
+
+// feeAccumulator is the per-block fee accumulator that gets all fees paid
+// in a block.
+type feeAccumulator struct {
+	balance quantity.Quantity
+}
+
+// AuthenticateAndPayFees authenticates the message signer and makes sure that
+// any gas fees are paid.
+//
+// This method modifies the common pool balance to transfer collected gas fees.
```

Comment:
common pool -> general pool?

Reply:
Oops, this comment is no longer correct, I'll fix it.